rainbow/leakage_models.py

```diff
@@ -0,0 +1,109 @@
+# This file is part of rainbow
+#
+# rainbow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+#
+# Copyright 2023 Jan Jancar
+
+"""
+This module implements some common leakage models, which
+can be used to trace memory addresses, memory values or registers.
+"""
+import abc
+import sys
+from typing import ClassVar, Literal
+
+if sys.version_info[0] < 3 or sys.version_info[0] == 3 and sys.version_info[1] < 10:
+    def hw(i):
+        return bin(i).count("1")
+else:
+    def hw(i):
+        return i.bit_count()
```

Comment:
Just to get an idea, did you compare performance with the table-based function that existed before? Looking at the implementation I guess the built-in `bit_count()` will be much faster, but the bin(i).count was definitely slower (which is ok though, I'm fine assuming everyone has python 3.10)

Reply:
Hmm, I did not compare performance, I think I looked at StackOverflow for performant ways to do HW computation in Python and these came up.